core/shared/src/main/scala/zio/Cause.scala

```diff
@@ -378,6 +421,17 @@ object Cause extends Serializable {
     }
   }
 
+  final case class Data private[Cause] (stackless: Boolean)
+
+  object Data {
+    final def apply(stackless: Boolean): Data =
```

Comment:
I think this function is implied by using a `case class` isn't it?

Reply:
We're trying to preserve the ability to augment `Data` with additional fields while preserving binary compatibility. So the thought from @neko-kai, if I understood it correctly, was that by overriding these methods we could then forward them to a new version and keep the old version private for compatibility. But I admit I am still learning about this so I am more just trying to follow the advice of others with more expertise at this point.